staging/src/k8s.io/api/resource/v1alpha2/types.go

```diff
@@ -192,11 +199,64 @@ type ResourceHandle struct {
 	// future, but not reduced.
 	// +optional
 	Data string `json:"data,omitempty" protobuf:"bytes,2,opt,name=data"`
+
+	// If StructuredData is set, then it needs to be used instead of Data.
+	//
+	// +optional
+	StructuredData *StructuredResourceHandle `json:"structuredData,omitempty" protobuf:"bytes,5,opt,name=structuredData"`
 }
 
 // ResourceHandleDataMaxSize represents the maximum size of resourceHandle.data.
 const ResourceHandleDataMaxSize = 16 * 1024
 
+// StructuredResourceHandle is the in-tree representation of the allocation result.
+type StructuredResourceHandle struct {
+	// VendorClassParameters are the per-claim configuration parameters
+	// from the resource class at the time that the claim was allocated.
+	//
+	// +optional
+	VendorClassParameters runtime.RawExtension `json:"vendorClassParameters,omitempty" protobuf:"bytes,1,opt,name=vendorClassParameters"`
+
+	// VendorClaimParameters are the per-claim configuration parameters
+	// from the resource claim parameters at the time that the claim was
+	// allocated.
+	//
+	// +optional
+	VendorClaimParameters runtime.RawExtension `json:"vendorClaimParameters,omitempty" protobuf:"bytes,2,opt,name=vendorClaimParameters"`
+
+	// NodeName is the name of the node providing the necessary resources.
+	// This mirrors the AllocationResult.AvailableOnNodes with a simpler
+	// type.
```

Comment:
Is this useful for users to know? What makes it simpler?

Reply:
Let me remove the second sentence. It's more like an explanation of the design than user-facing documentation.